packages/amplify-headless-interface/scripts/generateSchemas.ts

```diff
@@ -0,0 +1,49 @@
+import { getProgramFromFiles, buildGenerator, PartialArgs, Definition } from 'typescript-json-schema';
+import fs from 'fs-extra';
+import path from 'path';
+
+// paths are relative to the package root
+const typesSourceRoot = './src/interface/';
+const schemaFilesRoot = './schemas';
+
+// defines the type names and the paths to the TS files that define them
+const typeDefs: TypeDef[] = [
+  {
+    typeName: 'AddAuthRequest',
+    relativeSourcePaths: ['addAuth.ts'],
+  },
+  {
+    typeName: 'AddApiRequest',
+    relativeSourcePaths: ['addApi.ts'],
+  },
+  {
+    typeName: 'AddStorageRequest',
+    relativeSourcePaths: ['addStorage.ts'],
+  },
+];
+
+const schemaFileName = (typeName: string) => `${typeName}.schema.json`;
+const forceFlag = '--overwrite';
+const force = process.argv.includes(forceFlag);
+
+// schema generation settings. see https://www.npmjs.com/package/typescript-json-schema#command-line
+const settings: PartialArgs = {
+  required: true,
+};
+
+typeDefs.forEach(typeDef => {
+  const files = typeDef.relativeSourcePaths.map(p => path.resolve(path.join(typesSourceRoot, p)));
+  const typeSchema = buildGenerator(getProgramFromFiles(files), settings)!.getSchemaForSymbol(typeDef.typeName);
+  const version = (typeSchema.properties!.version as Definition).enum![0] as number;
+  const schemaFilePath = path.resolve(path.join(schemaFilesRoot, version.toString(), schemaFileName(typeDef.typeName)));
+  if (!force && fs.existsSync(schemaFilePath)) {
+    console.error(
+      `Schema version ${version} already exists for type ${typeDef.typeName}.\nThe interface version must be bumped after any changes.\nUse the ${forceFlag} flag to overwrite existing versions`,
+    );
+    process.exit(1);
+  }
+  fs.ensureFileSync(schemaFilePath);
```

Comment:
I believe writeFileSync does an ensureFileSync internally - hence you don't need to do it explcitly out here, please double check.

Reply:
tried it out and got a "no such file or directory" error